"Test events were not received" in Android Studio Artic Fox

After migrating to [Android Studio Artic Fox](https://developer.android.com/studio), when running unit tests from the IDE, it says "Test events were not received"

![image](https://user-images.githubusercontent.com/4595241/131808763-5844bcef-d559-4e4e-81e8-8c2743be886a.png)

Here is the output:
```
Executing tasks: [:app:cleanTestDebugUnitTest, :app:testDebugUnitTest] in project /Users/gmerino/Workspace/personal/TestsInRemote


> Configure project :
Here's Mirakle 1.4.3. All tasks will be executed on arc-prod-1.tuenti.io.
> Task :uploadToRemote
> Task :executeOnRemote
> Task :app:cleanTestDebugUnitTest
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :app:compileDebugAidl NO-SOURCE
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :app:generateDebugBuildConfig UP-TO-DATE
> Task :app:checkDebugAarMetadata UP-TO-DATE
> Task :app:generateDebugResValues UP-TO-DATE
> Task :app:generateDebugResources UP-TO-DATE
> Task :app:mergeDebugResources UP-TO-DATE
> Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
> Task :app:extractDeepLinksDebug UP-TO-DATE
> Task :app:processDebugMainManifest UP-TO-DATE
> Task :app:processDebugManifest UP-TO-DATE
> Task :app:processDebugManifestForPackage UP-TO-DATE
> Task :app:processDebugResources UP-TO-DATE
> Task :app:compileDebugKotlin
> Task :app:javaPreCompileDebug UP-TO-DATE
> Task :app:compileDebugJavaWithJavac UP-TO-DATE
> Task :app:bundleDebugClasses UP-TO-DATE
> Task :app:compileDebugUnitTestKotlin
> Task :app:preDebugUnitTestBuild UP-TO-DATE
> Task :app:javaPreCompileDebugUnitTest UP-TO-DATE
> Task :app:compileDebugUnitTestJavaWithJavac NO-SOURCE
> Task :app:processDebugJavaRes NO-SOURCE
> Task :app:processDebugUnitTestJavaRes NO-SOURCE
> Task :app:testDebugUnitTest
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/7.0.2/userguide/command_line_interface.html#sec:command_line_warnings
BUILD SUCCESSFUL in 2s
18 actionable tasks: 4 executed, 14 up-to-date
> Task :downloadInParallel
> Task :downloadFromRemote
> Task :mirakle
Total time : 5.592 secs
Task uploadToRemote took : 0.316 secs
Task executeOnRemote took : 3.027 secs
Task downloadFromRemote took : 0.393 secs
5 actionable tasks: 5 executed
10:06:51: Tasks execution finished ':app:cleanTestDebugUnitTest :app:testDebugUnitTest --tests "com.gmerino.testsinremote.ExampleUnitTest"'.
```
I've tried with different versions of Gradle and also use Junit4 & Junit5, but I wasn't able to find a workaround

**Steps to reproduce**
Create an empty project in Android Studio, it will create a class called ExampleUnitTest. Try executing that test when mirakle is configured.

**IntelliJ**
I've also reproduced it in IntelliJ IDEA 2021.2.1 with the same STRs

Hmm sorry, I thought it was going well, but checking the logs I'm seeing

```
> Task :uploadToRemote
> Task :executeOnRemote
FAILURE: Build failed with an exception.
* Where:
Initialization script '/Users/gmerino/Workspace/android-messenger/mirakle_init_scripts/ijresolvers1.gradle' line: 416
* What went wrong:
Could not compile initialization script '/Users/gmerino/Workspace/android-messenger/mirakle_init_scripts/ijresolvers1.gradle'.
> startup failed:
  initialization script '/Users/gmerino/Workspace/android-messenger/mirakle_init_scripts/ijresolvers1.gradle': 416: unable to resolve class com.intellij.openapi.externalSystem.rt.execution.ForkedDebuggerHelper
   @ line 416, column 1.
     import com.intellij.openapi.externalSystem.rt.execution.ForkedDebuggerHelper    
     ^
  
  1 error
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 802ms
```

does that long ring any bells?

I've tried with that version, here is the log for a failing test:

```
> Task :web:testMovistarESDebugUnitTest FAILED
<ijLog><event type='beforeTest'><ijLogEol/>  <test id='4.3' parentId='4.2'><ijLogEol/>    <descriptor name='handler should return not installed app and market urls for given app token' className='com.tuenti.web.bridge.GetAppMetadataTest' /><ijLogEol/>  </test><ijLogEol/></event></ijLog>
<ijLog><event type='afterTest'><ijLogEol
3 tests completed, 2 failed
/>  <test id='4.3' parentId='4.2'><ijLogEol/>    <descriptor name='handler should return not installed app and market urls for given app token' className='com.tuenti.web.bridge.GetAppMetadataTest' /><ijLogEol/>    <result resultType='FAILURE' startTime='1635155570726' endTime='1635155570746'><errorMsg><![CDATA[ZXhwZWN0ZWQ6PFtHRVRfQVBQX01FVEFEQVRBXT4gYnV0IHdhczo8W0VSUk9SXT4=]]></errorMsg><stackTrace><![CDATA[b3JnLmp1bml0LkNvbXBhcmlzb25GYWlsdXJlOiBleHBlY3RlZDo8W0dFVF9BUFBfTUVUQURBVEFdPiBidXQgd2FzOjxbRVJST1JdPgoJYXQgb3JnLmp1bml0LkFzc2VydC5hc3NlcnRFcXVhbHMoQXNzZXJ0LmphdmE6MTE1KQoJYXQgb3JnLmp1bml0LkFzc2VydC5hc3NlcnRFcXVhbHMoQXNzZXJ0LmphdmE6MTQ0KQoJYXQgY29tLnR1ZW50aS53ZWIuYnJpZGdlLk1lc3NhZ2VIYW5kbGVyc1Rlc3RzS3QuZXF1YWxzVG8oTWVzc2FnZUhhbmRsZXJzVGVzdHMua3Q6NTkpCglhdCBjb20udHVlbnRpLndlYi5icmlkZ2UuTWVzc2FnZUhhbmRsZXJzVGVzdHNLdC5lcXVhbHNUbyRkZWZhdWx0KE1lc3NhZ2VIYW5kbGVyc1Rlc3RzLmt0OjU4KQoJYXQgY29tLnR1ZW50aS53ZWIuYnJpZGdlLkdldEFwcE1ldGFkYXRhVGVzdC52ZXJpZnlSZXNwb25zZShHZXRBcHBNZXRhZGF0YVRlc3Qua3Q6NzApCglhdCBjb20udHVlbnRpLndlYi5icmlkZ2UuR2V0QXBwTWV0YWRhdGFUZXN0LmhhbmRsZXIgc2hvdWxkIHJldHVybiBub3QgaW5zdGFsbGVkIGFwcCBhbmQgbWFya2V0IHVybHMgZm9yIGdpdmVuIGFwcCB0b2tlbihHZXRBcHBNZXRhZGF0YVRlc3Qua3Q6NDUpCglhdCBqYXZhLmJhc2UvamRrLmludGVybmFsLnJlZmxlY3QuTmF0aXZlTWV0aG9kQWNjZXNzb3JJbXBsLmludm9rZTAoTmF0aXZlIE1ldGhvZCkKCWF0IGphdmEuYmFzZS9qZGsuaW50ZXJuYWwucmVmbGVjdC5OYXRpdmVNZXRob2RBY2Nlc3NvckltcGwuaW52b2tlKE5hdGl2ZU1ldGhvZEFjY2Vzc29ySW1wbC5qYXZhOjYyKQoJYXQgamF2YS5iYXNlL2pkay5pbnRlcm5hbC5yZWZsZWN0LkRlbGVnYXRpbmdNZXRob2RBY2Nlc3NvckltcGwuaW52b2tlKERlbGVnYXRpbmdNZXRob2RBY2Nlc3NvckltcGwuamF2YTo0MykKCWF0IGphdmEuYmFzZS9qYXZhLmxhbmcucmVmbGVjdC5NZXRob2QuaW52b2tlKE1ldGhvZC5qYXZhOjU2NikKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLm1vZGVsLkZyYW1ld29ya01ldGhvZCQxLnJ1blJlZmxlY3RpdmVDYWxsKEZyYW1ld29ya01ldGhvZC5qYXZhOjUwKQoJYXQgb3JnLmp1bml0LmludGVybmFsLnJ1bm5lcnMubW9kZWwuUmVmbGVjdGl2ZUNhbGxhYmxlLnJ1bihSZWZsZWN0aXZlQ2FsbGFibGUuamF2YToxMikKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLm1vZGVsLkZyYW1ld29ya01ldGhvZC5pbnZva2VFeHBsb3NpdmVseShGcmFtZXdvcmtNZXRob2QuamF2YTo0NykKCWF0IG9yZy5qdW5pdC5pbnRlcm5hbC5ydW5uZXJzLnN0YXRlbWVudHMuSW52b2tlTWV0aG9kLmV2YWx1YXRlKEludm9rZU1ldGhvZC5qYXZhOjE3KQoJYXQgb3JnLmp1bml0LmludGVybmFsLnJ1bm5lcnMuc3RhdGVtZW50cy5SdW5CZWZvcmVzLmV2YWx1YXRlKFJ1bkJlZm9yZXMuamF2YToyNikKCWF0IG9yZy5qdW5pdC5pbnRlcm5hbC5ydW5uZXJzLnN0YXRlbWVudHMuUnVuQWZ0ZXJzLmV2YWx1YXRlKFJ1bkFmdGVycy5qYXZhOjI3KQoJYXQgb3JnLmp1bml0LnJ1bGVzLlRlc3RXYXRjaGVyJDEuZXZhbHVhdGUoVGVzdFdhdGNoZXIuamF2YTo1NSkKCWF0IGNvbS50dWVudGkubWVzc2VuZ2VyLnRlc3QucnVsZXMuc3RhdGVtZW50LlRlc3RUaW1lb3V0U3RhdGVtZW50LmV2YWx1YXRlKFRlc3RUaW1lb3V0U3RhdGVtZW50LmphdmE6NDIpCglhdCBvcmcuanVuaXQucnVsZXMuUnVuUnVsZXMuZXZhbHVhdGUoUnVuUnVsZXMuamF2YToyMCkKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLlBhcmVudFJ1bm5lci5ydW5MZWFmKFBhcmVudFJ1bm5lci5qYXZhOjMyNSkKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLkJsb2NrSlVuaXQ0Q2xhc3NSdW5uZXIucnVuQ2hpbGQoQmxvY2tKVW5pdDRDbGFzc1J1bm5lci5qYXZhOjc4KQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMuQmxvY2tKVW5pdDRDbGFzc1J1bm5lci5ydW5DaGlsZChCbG9ja0pVbml0NENsYXNzUnVubmVyLmphdmE6NTcpCglhdCBvcmcuanVuaXQucnVubmVycy5QYXJlbnRSdW5uZXIkMy5ydW4oUGFyZW50UnVubmVyLmphdmE6MjkwKQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMuUGFyZW50UnVubmVyJDEuc2NoZWR1bGUoUGFyZW50UnVubmVyLmphdmE6NzEpCglhdCBvcmcuanVuaXQucnVubmVycy5QYXJlbnRSdW5uZXIucnVuQ2hpbGRyZW4oUGFyZW50UnVubmVyLmphdmE6Mjg4KQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMuUGFyZW50UnVubmVyLmFjY2VzcyQwMDAoUGFyZW50UnVubmVyLmphdmE6NTgpCglhdCBvcmcuanVuaXQucnVubmVycy5QYXJlbnRSdW5uZXIkMi5ldmFsdWF0ZShQYXJlbnRSdW5uZXIuamF2YToyNjgpCglhdCBvcmcuanVuaXQucnVubmVycy5QYXJlbnRSdW5uZXIucnVuKFBhcmVudFJ1bm5lci5qYXZhOjM2MykKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3RpbmcuanVuaXQuSlVuaXRUZXN0Q2xhc3NFeGVjdXRvci5ydW5UZXN0Q2xhc3MoSlVuaXRUZXN0Q2xhc3NFeGVjdXRvci5qYXZhOjExMCkKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3RpbmcuanVuaXQuSlVuaXRUZXN0Q2xhc3NFeGVjdXRvci5leGVjdXRlKEpVbml0VGVzdENsYXNzRXhlY3V0b3IuamF2YTo1OCkKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3RpbmcuanVuaXQuSlVuaXRUZXN0Q2xhc3NFeGVjdXRvci5leGVjdXRlKEpVbml0VGVzdENsYXNzRXhlY3V0b3IuamF2YTozOCkKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3RpbmcuanVuaXQuQWJzdHJhY3RKVW5pdFRlc3RDbGFzc1Byb2Nlc3Nvci5wcm9jZXNzVGVzdENsYXNzKEFic3RyYWN0SlVuaXRUZXN0Q2xhc3NQcm9jZXNzb3IuamF2YTo2MikKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3RpbmcuU3VpdGVUZXN0Q2xhc3NQcm9jZXNzb3IucHJvY2Vzc1Rlc3RDbGFzcyhTdWl0ZVRlc3RDbGFzc1Byb2Nlc3Nvci5qYXZhOjUxKQoJYXQgamF2YS5iYXNlL2pkay5pbnRlcm5hbC5yZWZsZWN0Lk5hdGl2ZU1ldGhvZEFjY2Vzc29ySW1wbC5pbnZva2UwKE5hdGl2ZSBNZXRob2QpCglhdCBqYXZhLmJhc2UvamRrLmludGVybmFsLnJlZmxlY3QuTmF0aXZlTWV0aG9kQWNjZXNzb3JJbXBsLmludm9rZShOYXRpdmVNZXRob2RBY2Nlc3NvckltcGwuamF2YTo2MikKCWF0IGphdmEuYmFzZS9qZGsuaW50ZXJuYWwucmVmbGVjdC5EZWxlZ2F0aW5nTWV0aG9kQWNjZXNzb3JJbXBsLmludm9rZShEZWxlZ2F0aW5nTWV0aG9kQWNjZXNzb3JJbXBsLmphdmE6NDMpCglhdCBqYXZhLmJhc2UvamF2YS5sYW5nLnJlZmxlY3QuTWV0aG9kLmludm9rZShNZXRob2QuamF2YTo1NjYpCglhdCBvcmcuZ3JhZGxlLmludGVybmFsLmRpc3BhdGNoLlJlZmxlY3Rpb25EaXNwYXRjaC5kaXNwYXRjaChSZWZsZWN0aW9uRGlzcGF0Y2guamF2YTozNikKCWF0IG9yZy5ncmFkbGUuaW50ZXJuYWwuZGlzcGF0Y2guUmVmbGVjdGlvbkRpc3BhdGNoLmRpc3BhdGNoKFJlZmxlY3Rpb25EaXNwYXRjaC5qYXZhOjI0KQoJYXQgb3JnLmdyYWRsZS5pbnRlcm5hbC5kaXNwYXRjaC5Db250ZXh0Q2xhc3NMb2FkZXJEaXNwYXRjaC5kaXNwYXRjaChDb250ZXh0Q2xhc3NMb2FkZXJEaXNwYXRjaC5qYXZhOjMzKQoJYXQgb3JnLmdyYWRsZS5pbnRlcm5hbC5kaXNwYXRjaC5Qcm94eURpc3BhdGNoQWRhcHRlciREaXNwYXRjaGluZ0ludm9jYXRpb25IYW5kbGVyLmludm9rZShQcm94eURpc3BhdGNoQWRhcHRlci5qYXZhOjk0KQoJYXQgY29tLnN1bi5wcm94eS4kUHJveHkyLnByb2Nlc3NUZXN0Q2xhc3MoVW5rbm93biBTb3VyY2UpCglhdCBvcmcuZ3JhZGxlLmFwaS5pbnRlcm5hbC50YXNrcy50ZXN0aW5nLndvcmtlci5UZXN0V29ya2VyJDIucnVuKFRlc3RXb3JrZXIuamF2YToxNzYpCglhdCBvcmcuZ3JhZGxlLmFwaS5pbnRlcm5hbC50YXNrcy50ZXN0aW5nLndvcmtlci5UZXN0V29ya2VyLmV4ZWN1dGVBbmRNYWludGFpblRocmVhZE5hbWUoVGVzdFdvcmtlci5qYXZhOjEyOSkKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3Rpbmcud29ya2VyLlRlc3RXb3JrZXIuZXhlY3V0ZShUZXN0V29ya2VyLmphdmE6MTAwKQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy53b3JrZXIuVGVzdFdvcmtlci5leGVjdXRlKFRlc3RXb3JrZXIuamF2YTo2MCkKCWF0IG9yZy5ncmFkbGUucHJvY2Vzcy5pbnRlcm5hbC53b3JrZXIuY2hpbGQuQWN0aW9uRXhlY3V0aW9uV29ya2VyLmV4ZWN1dGUoQWN0aW9uRXhlY3V0aW9uV29ya2VyLmphdmE6NTYpCglhdCBvcmcuZ3JhZGxlLnByb2Nlc3MuaW50ZXJuYWwud29ya2VyLmNoaWxkLlN5c3RlbUFwcGxpY2F0aW9uQ2xhc3NMb2FkZXJXb3JrZXIuY2FsbChTeXN0ZW1BcHBsaWNhdGlvbkNsYXNzTG9hZGVyV29ya2VyLmphdmE6MTMzKQoJYXQgb3JnLmdyYWRsZS5wcm9jZXNzLmludGVybmFsLndvcmtlci5jaGlsZC5TeXN0ZW1BcHBsaWNhdGlvbkNsYXNzTG9hZGVyV29ya2VyLmNhbGwoU3lzdGVtQXBwbGljYXRpb25DbGFzc0xvYWRlcldvcmtlci5qYXZhOjcxKQoJYXQgd29ya2VyLm9yZy5ncmFkbGUucHJvY2Vzcy5pbnRlcm5hbC53b3JrZXIuR3JhZGxlV29ya2VyTWFpbi5ydW4oR3JhZGxlV29ya2VyTWFpbi5qYXZhOjY5KQoJYXQgd29ya2VyLm9yZy5ncmFkbGUucHJvY2Vzcy5pbnRlcm5hbC53b3JrZXIuR3JhZGxlV29ya2VyTWFpbi5tYWluKEdyYWRsZVdvcmtlck1haW4uamF2YTo3NCkK]]></stackTrace><ijLogEol/>      <failureType>comparison</failureType><expected><![CDATA[R0VUX0FQUF9NRVRBREFUQQ==]]></expected><actual><![CDATA[RVJST1I=]]></actual><ijLogEol/>    </result><ijLogEol/>  </test><ijLogEol/></event></ijLog>
com.tuenti.web.bridge.GetAppMetadataTest > handler should return not installed app and market urls for given app token FAILED
    org.junit.ComparisonFailure at GetAppMetadataTest.kt:70


<ijLog><event type='afterTest'><ijLogEol/>  <test id='4.5' parentId='4.2'><ijLogEol/>    <descriptor name='handler should return installed app and market urls for given app token' className='com.tuenti.web.bridge.GetAppMetadataTest' /><ijLogEol/>    <result resultType='FAILURE' startTime='1635155570746' endTime='1635155570749'><errorMsg><![CDATA[ZXhwZWN0ZWQ6PFtHRVRfQVBQX01FVEFEQVRBXT4gYnV0IHdhczo8W0VSUk9SXT4=]]></errorMsg><stackTrace><![CDATA[b3JnLmp1bml0LkNvbXBhcmlzb25GYWlsdXJlOiBleHBlY3RlZDo8W0dFVF9BUFBfTUVUQURBVEFdPiBidXQgd2FzOjxbRVJST1JdPgoJYXQgb3JnLmp1bml0LkFzc2VydC5hc3NlcnRFcXVhbHMoQXNzZXJ0LmphdmE6MTE1KQoJYXQgb3JnLmp1bml0LkFzc2VydC5hc3NlcnRFcXVhbHMoQXNzZXJ0LmphdmE6MTQ0KQoJYXQgY29tLnR1ZW50aS53ZWIuYnJpZGdlLk1lc3NhZ2VIYW5kbGVyc1Rlc3RzS3QuZXF1YWxzVG8oTWVzc2FnZUhhbmRsZXJzVGVzdHMua3Q6NTkpCglhdCBjb20udHVlbnRpLndlYi5icmlkZ2UuTWVzc2FnZUhhbmRsZXJzVGVzdHNLdC5lcXVhbHNUbyRkZWZhdWx0KE1lc3NhZ2VIYW5kbGVyc1Rlc3RzLmt0OjU4KQoJYXQgY29tLnR1ZW50aS53ZWIuYnJpZGdlLkdldEFwcE1ldGFkYXRhVGVzdC52ZXJpZnlSZXNwb25zZShHZXRBcHBNZXRhZGF0YVRlc3Qua3Q6NzApCglhdCBjb20udHVlbnRpLndlYi5icmlkZ2UuR2V0QXBwTWV0YWRhdGFUZXN0LmhhbmRsZXIgc2hvdWxkIHJldHVybiBpbnN0YWxsZWQgYXBwIGFuZCBtYXJrZXQgdXJscyBmb3IgZ2l2ZW4gYXBwIHRva2VuKEdldEFwcE1ldGFkYXRhVGVzdC5rdDozNykKCWF0IGphdmEuYmFzZS9qZGsuaW50ZXJuYWwucmVmbGVjdC5OYXRpdmVNZXRob2RBY2Nlc3NvckltcGwuaW52b2tlMChOYXRpdmUgTWV0aG9kKQoJYXQgamF2YS5iYXNlL2pkay5pbnRlcm5hbC5yZWZsZWN0Lk5hdGl2ZU1ldGhvZEFjY2Vzc29ySW1wbC5pbnZva2UoTmF0aXZlTWV0aG9kQWNjZXNzb3JJbXBsLmphdmE6NjIpCglhdCBqYXZhLmJhc2UvamRrLmludGVybmFsLnJlZmxlY3QuRGVsZWdhdGluZ01ldGhvZEFjY2Vzc29ySW1wbC5pbnZva2UoRGVsZWdhdGluZ01ldGhvZEFjY2Vzc29ySW1wbC5qYXZhOjQzKQoJYXQgamF2YS5iYXNlL2phdmEubGFuZy5yZWZsZWN0Lk1ldGhvZC5pbnZva2UoTWV0aG9kLmphdmE6NTY2KQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMubW9kZWwuRnJhbWV3b3JrTWV0aG9kJDEucnVuUmVmbGVjdGl2ZUNhbGwoRnJhbWV3b3JrTWV0aG9kLmphdmE6NTApCglhdCBvcmcuanVuaXQuaW50ZXJuYWwucnVubmVycy5tb2RlbC5SZWZsZWN0aXZlQ2FsbGFibGUucnVuKFJlZmxlY3RpdmVDYWxsYWJsZS5qYXZhOjEyKQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMubW9kZWwuRnJhbWV3b3JrTWV0aG9kLmludm9rZUV4cGxvc2l2ZWx5KEZyYW1ld29ya01ldGhvZC5qYXZhOjQ3KQoJYXQgb3JnLmp1bml0LmludGVybmFsLnJ1bm5lcnMuc3RhdGVtZW50cy5JbnZva2VNZXRob2QuZXZhbHVhdGUoSW52b2tlTWV0aG9kLmphdmE6MTcpCglhdCBvcmcuanVuaXQuaW50ZXJuYWwucnVubmVycy5zdGF0ZW1lbnRzLlJ1bkJlZm9yZXMuZXZhbHVhdGUoUnVuQmVmb3Jlcy5qYXZhOjI2KQoJYXQgb3JnLmp1bml0LmludGVybmFsLnJ1bm5lcnMuc3RhdGVtZW50cy5SdW5BZnRlcnMuZXZhbHVhdGUoUnVuQWZ0ZXJzLmphdmE6MjcpCglhdCBvcmcuanVuaXQucnVsZXMuVGVzdFdhdGNoZXIkMS5ldmFsdWF0ZShUZXN0V2F0Y2hlci5qYXZhOjU1KQoJYXQgY29tLnR1ZW50aS5tZXNzZW5nZXIudGVzdC5ydWxlcy5zdGF0ZW1lbnQuVGVzdFRpbWVvdXRTdGF0ZW1lbnQuZXZhbHVhdGUoVGVzdFRpbWVvdXRTdGF0ZW1lbnQuamF2YTo0MikKCWF0IG9yZy5qdW5pdC5ydWxlcy5SdW5SdWxlcy5ldmFsdWF0ZShSdW5SdWxlcy5qYXZhOjIwKQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMuUGFyZW50UnVubmVyLnJ1bkxlYWYoUGFyZW50UnVubmVyLmphdmE6MzI1KQoJYXQgb3JnLmp1bml0LnJ1bm5lcnMuQmxvY2tKVW5pdDRDbGFzc1J1bm5lci5ydW5DaGlsZChCbG9ja0pVbml0NENsYXNzUnVubmVyLmphdmE6NzgpCglhdCBvcmcuanVuaXQucnVubmVycy5CbG9ja0pVbml0NENsYXNzUnVubmVyLnJ1bkNoaWxkKEJsb2NrSlVuaXQ0Q2xhc3NSdW5uZXIuamF2YTo1NykKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLlBhcmVudFJ1bm5lciQzLnJ1bihQYXJlbnRSdW5uZXIuamF2YToyOTApCglhdCBvcmcuanVuaXQucnVubmVycy5QYXJlbnRSdW5uZXIkMS5zY2hlZHVsZShQYXJlbnRSdW5uZXIuamF2YTo3MSkKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLlBhcmVudFJ1bm5lci5ydW5DaGlsZHJlbihQYXJlbnRSdW5uZXIuamF2YToyODgpCglhdCBvcmcuanVuaXQucnVubmVycy5QYXJlbnRSdW5uZXIuYWNjZXNzJDAwMChQYXJlbnRSdW5uZXIuamF2YTo1OCkKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLlBhcmVudFJ1bm5lciQyLmV2YWx1YXRlKFBhcmVudFJ1bm5lci5qYXZhOjI2OCkKCWF0IG9yZy5qdW5pdC5ydW5uZXJzLlBhcmVudFJ1bm5lci5ydW4oUGFyZW50UnVubmVyLmphdmE6MzYzKQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy5qdW5pdC5KVW5pdFRlc3RDbGFzc0V4ZWN1dG9yLnJ1blRlc3RDbGFzcyhKVW5pdFRlc3RDbGFzc0V4ZWN1dG9yLmphdmE6MTEwKQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy5qdW5pdC5KVW5pdFRlc3RDbGFzc0V4ZWN1dG9yLmV4ZWN1dGUoSlVuaXRUZXN0Q2xhc3NFeGVjdXRvci5qYXZhOjU4KQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy5qdW5pdC5KVW5pdFRlc3RDbGFzc0V4ZWN1dG9yLmV4ZWN1dGUoSlVuaXRUZXN0Q2xhc3NFeGVjdXRvci5qYXZhOjM4KQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy5qdW5pdC5BYnN0cmFjdEpVbml0VGVzdENsYXNzUHJvY2Vzc29yLnByb2Nlc3NUZXN0Q2xhc3MoQWJzdHJhY3RKVW5pdFRlc3RDbGFzc1Byb2Nlc3Nvci5qYXZhOjYyKQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy5TdWl0ZVRlc3RDbGFzc1Byb2Nlc3Nvci5wcm9jZXNzVGVzdENsYXNzKFN1aXRlVGVzdENsYXNzUHJvY2Vzc29yLmphdmE6NTEpCglhdCBqYXZhLmJhc2UvamRrLmludGVybmFsLnJlZmxlY3QuTmF0aXZlTWV0aG9kQWNjZXNzb3JJbXBsLmludm9rZTAoTmF0aXZlIE1ldGhvZCkKCWF0IGphdmEuYmFzZS9qZGsuaW50ZXJuYWwucmVmbGVjdC5OYXRpdmVNZXRob2RBY2Nlc3NvckltcGwuaW52b2tlKE5hdGl2ZU1ldGhvZEFjY2Vzc29ySW1wbC5qYXZhOjYyKQoJYXQgamF2YS5iYXNlL2pkay5pbnRlcm5hbC5yZWZsZWN0LkRlbGVnYXRpbmdNZXRob2RBY2Nlc3NvckltcGwuaW52b2tlKERlbGVnYXRpbmdNZXRob2RBY2Nlc3NvckltcGwuamF2YTo0MykKCWF0IGphdmEuYmFzZS9qYXZhLmxhbmcucmVmbGVjdC5NZXRob2QuaW52b2tlKE1ldGhvZC5qYXZhOjU2NikKCWF0IG9yZy5ncmFkbGUuaW50ZXJuYWwuZGlzcGF0Y2guUmVmbGVjdGlvbkRpc3BhdGNoLmRpc3BhdGNoKFJlZmxlY3Rpb25EaXNwYXRjaC5qYXZhOjM2KQoJYXQgb3JnLmdyYWRsZS5pbnRlcm5hbC5kaXNwYXRjaC5SZWZsZWN0aW9uRGlzcGF0Y2guZGlzcGF0Y2goUmVmbGVjdGlvbkRpc3BhdGNoLmphdmE6MjQpCglhdCBvcmcuZ3JhZGxlLmludGVybmFsLmRpc3BhdGNoLkNvbnRleHRDbGFzc0xvYWRlckRpc3BhdGNoLmRpc3BhdGNoKENvbnRleHRDbGFzc0xvYWRlckRpc3BhdGNoLmphdmE6MzMpCglhdCBvcmcuZ3JhZGxlLmludGVybmFsLmRpc3BhdGNoLlByb3h5RGlzcGF0Y2hBZGFwdGVyJERpc3BhdGNoaW5nSW52b2NhdGlvbkhhbmRsZXIuaW52b2tlKFByb3h5RGlzcGF0Y2hBZGFwdGVyLmphdmE6OTQpCglhdCBjb20uc3VuLnByb3h5LiRQcm94eTIucHJvY2Vzc1Rlc3RDbGFzcyhVbmtub3duIFNvdXJjZSkKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3Rpbmcud29ya2VyLlRlc3RXb3JrZXIkMi5ydW4oVGVzdFdvcmtlci5qYXZhOjE3NikKCWF0IG9yZy5ncmFkbGUuYXBpLmludGVybmFsLnRhc2tzLnRlc3Rpbmcud29ya2VyLlRlc3RXb3JrZXIuZXhlY3V0ZUFuZE1haW50YWluVGhyZWFkTmFtZShUZXN0V29ya2VyLmphdmE6MTI5KQoJYXQgb3JnLmdyYWRsZS5hcGkuaW50ZXJuYWwudGFza3MudGVzdGluZy53b3JrZXIuVGVzdFdvcmtlci5leGVjdXRlKFRlc3RXb3JrZXIuamF2YToxMDApCglhdCBvcmcuZ3JhZGxlLmFwaS5pbnRlcm5hbC50YXNrcy50ZXN0aW5nLndvcmtlci5UZXN0V29ya2VyLmV4ZWN1dGUoVGVzdFdvcmtlci5qYXZhOjYwKQoJYXQgb3JnLmdyYWRsZS5wcm9jZXNzLmludGVybmFsLndvcmtlci5jaGlsZC5BY3Rpb25FeGVjdXRpb25Xb3JrZXIuZXhlY3V0ZShBY3Rpb25FeGVjdXRpb25Xb3JrZXIuamF2YTo1NikKCWF0IG9yZy5ncmFkbGUucHJvY2Vzcy5pbnRlcm5hbC53b3JrZXIuY2hpbGQuU3lzdGVtQXBwbGljYXRpb25DbGFzc0xvYWRlcldvcmtlci5jYWxsKFN5c3RlbUFwcGxpY2F0aW9uQ2xhc3NMb2FkZXJXb3JrZXIuamF2YToxMzMpCglhdCBvcmcuZ3JhZGxlLnByb2Nlc3MuaW50ZXJuYWwud29ya2VyLmNoaWxkLlN5c3RlbUFwcGxpY2F0aW9uQ2xhc3NMb2FkZXJXb3JrZXIuY2FsbChTeXN0ZW1BcHBsaWNhdGlvbkNsYXNzTG9hZGVyV29ya2VyLmphdmE6NzEpCglhdCB3b3JrZXIub3JnLmdyYWRsZS5wcm9jZXNzLmludGVybmFsLndvcmtlci5HcmFkbGVXb3JrZXJNYWluLnJ1bihHcmFkbGVXb3JrZXJNYWluLmphdmE6NjkpCglhdCB3b3JrZXIub3JnLmdyYWRsZS5wcm9jZXNzLmludGVybmFsLndvcmtlci5HcmFkbGVXb3JrZXJNYWluLm1haW4oR3JhZGxlV29ya2VyTWFpbi5qYXZhOjc0KQo=]]></stackTrace><ijLogEol/>      <failureType>comparison</failureType><expected><![CDATA[R0VUX0FQUF9NRVRBREFUQQ==]]></expected><actual><![CDATA[RVJST1I=]]></actual><ijLogEol/>    </result><ijLogEol/>  </test><ijLogEol/></event></ijLog>
com.tuenti.web.bridge.GetAppMetadataTest > handler should return installed app and market urls for given app token FAILED
    org.junit.ComparisonFailure at GetAppMetadataTest.kt:70
<ijLog><event type='afterSuite'><ijLogEol
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':web:testMovistarESDebugUnitTest'.
/>  <test id=':web:testMovistarESDebugUnitTest' parentId=''><ijLogEol/>    <descriptor name='Gradle Test Run :web:testMovistarESDebugUnitTest' className='' /><ijLogEol/>    <result resultType='FAILURE' startTime='1635155569925' endTime='1635155570837'><ijLogEol/>      <failureType>error</failureType><ijLogEol/>    </result><ijLogEol/>  </test><ijLogEol/></event></ijLog>
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
> There were failing tests. See the report at: file:/Users/gmerino/Workspace/android-messenger/web/build/reports/tests/testMovistarESDebugUnitTest/index.html
* Try:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
See https:> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
//docs.gradle.org/7.3-rc-2/userguide/command_line_interface.html#sec:command_line_warnings
259 actionable tasks: 4 executed, 255 up-to-date
BUILD FAILED in 14s
> Task :downloadInParallel
> Task :downloadFromRemote
> Task :mirakle FAILED
Total time : 22.288 secs
Task uploadToRemote took : 0.726 secs
Task executeOnRemote took : 15.39 secs
Task downloadFromRemote took : 0.834 secs
9 actionable tasks: 9 executed
11:52:54: Tasks execution finished ':web:cleanTestMovistarESDebugUnitTest :web:testMovistarESDebugUnitTest --tests "com.tuenti.web.bridge.GetAppMetadataTest"'.
```

and the IDE keeps "executing" the test
![image](https://user-images.githubusercontent.com/4595241/138675113-833817e2-8e04-4072-bdfd-e9eca18a6cc9.png)

--- a/plugin/src/main/kotlin/Mirakle.kt
+++ b/plugin/src/main/kotlin/Mirakle.kt
@@ -160,16 +160,20 @@ open class Mirakle : Plugin<Gradle> {
 
                     isIgnoreExitValue = true
 
-                    standardOutput = modifyOutputStream(
-                            standardOutput ?: System.out,
-                            "${config.remoteFolder}/${gradlewRoot.name}",
-                            gradlewRoot.path
-                    )
-                    errorOutput = modifyOutputStream(
-                            errorOutput ?: System.err,
-                            "${config.remoteFolder}/${gradlewRoot.name}",
-                            gradlewRoot.path
-                    )
+                    // disable remote stream modification since it breaks Idea and Android Studio test result processing
+                    // https://github.com/Adambl4/mirakle/issues/83
+                    if (!gradle.containsIjTestInit()) {
+                        standardOutput = modifyOutputStream(
+                                standardOutput ?: System.out,
+                                "${config.remoteFolder}/${gradlewRoot.name}",
+                                gradlewRoot.path
+                        )
+                        errorOutput = modifyOutputStream(
+                                errorOutput ?: System.err,
+                                "${config.remoteFolder}/${gradlewRoot.name}",
+                                gradlewRoot.path
+                        )
+                    }
                 }.mustRunAfter(upload) as Exec
 
                 val download = project.task<Exec>("downloadFromRemote") {
@@ -690,4 +694,9 @@ fun mergeStartParamsWithProperties(
 
         addPropertiesToStartParams(mirakleProperties + mirakleLocalProperties, this)
     }
+}
+
+
+fun Gradle.containsIjTestInit() = startParameter.initScripts.any {
+    it.name == "ijtestinit.gradle"
 }

